### DISCUSSION TITLE
Trouble restoring seqbac file

### DISCUSSION THREAD

**Discussion by johnstein-storewise** [datalust/seq-tickets#1737] [Help]
We are trying to set up a new instance of Seq (from docker 5.1.3344 to docker latest). It looks like the "seqsvr restore" is prompting to overwrite from "Init Script". Here is snip from the log, not sure how to handle:

running init script seqbackuprestore.sh...
[17:18:28 INF] Opening metastore "/data/Documents/metastore.flare"
This will clear all metadata (users, signals, ...) from the current instance and replace it with the contents of backup /data/Backups/backup.seqbac.

Press `y` to continue, or any other key to cancel.
[17:18:28 INF] Closing metastore
The backup could not be restored; an exception occurred.
Exception message: Cannot read keys when either application does not have a console or when console input has been redirected. Try 
Console.Read.

Here is the content from seqbackuprestore.sh: seqsvr restore --secret-key="seqkey" --backup="/data/Backups/backup.seqbac" --enable-apps

Thanks,
John

**Comment by nblumhardt** [comment]
Hi John, thanks for reaching out.

You can pass `-y` on the command line to `seqsvr restore` to skip the prompt. Let me know if this does the trick!

Best regards,
Nick

**Reply by johnstein-storewise** [reply]
Nick,

That did the trick, but there seem to be another issue when restoring:

running init script seqbackuprestore.sh...
[23:21:23 INF] Opening metastore "/data/Documents/metastore.flare"
[23:21:23 INF] Restoring from "/data/Backups/backup.seqbac"
[23:21:23 INF] Decrypting content
[23:21:23 INF] Extracting content
[23:21:23 INF] Reading manifest
[23:21:23 INF] Closing metastore
The backup could not be restored; an exception occurred.
Exception message: Cannot deserialize the current JSON object (e.g. {"name":"value"}) into type 'System.Version' because the type requires a JSON string value to deserialize correctly.
To fix this error either change the JSON to a JSON string value or change the deserialized type so that it is a normal .NET type (e.g. not a primitive type like integer, not a collection type like an array or List<T>) that can be deserialized from a JSON object. JsonObjectAttribute can also be added to the type to force it to deserialize from a JSON object.
Path 'SeqVersion.Major', line 1, position 68.

Thanks,
John

**Reply by nblumhardt** [reply]
Thanks for the update, John.

I think this may be caused by the mismatch between the original and new Seq versions (5.1 vs 2022.1).

Init scripts aren't supported in 5.1, so upgrading the old instance in-place before taking a new backup would be the simplest way to approach it.

The other option might be to restore the backup on a temporary machine using a local Docker instance and the 5.1.3364 container, before upgrading on the temporary instance, and taking a new backup there.

Let me know if either of the above sounds like a fit!

Best regards,
Nick
